plugins/templates/getTemplateDetails.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const boom = require('@hapi/boom');
+const joi = require('joi');
+const schema = require('screwdriver-data-schema');
+const getSchema = schema.models.template.get;
+const idSchema = schema.models.template.extract('id');
+
+module.exports = () => ({
+    method: 'GET',
+    path: '/templates/{id}',
+    options: {
+        description: 'Get a single template',
+        notes: 'Returns a template record',
+        tags: ['api', 'templates'],
+        auth: {
+            strategies: ['token'],
+            scope: ['build', 'template', 'pipeline']
+        },
+
+        handler: async (request, h) => {
+            const { templateFactory  } = request.server.app;
+
+            return templateFactory
+                .get(request.params.id)
+                .then(async templateModel => {
+                    if (!templateModel) {
+                        throw boom.notFound('Template does not exist');
+                    }
+
+                    if (Array.isArray(templateFactory.environment)) {
+                        const data = await templateFactory.toJsonWithSteps();
```

Comment:
Hmm,  2 things:

1. I dont think `templateFactory` has `toJsonWithSteps` method, where did you this? https://github.com/screwdriver-cd/models/blob/9591cc5e19/lib/templateFactory.js